teamapps-ux/src/main/java/org/teamapps/ux/session/TeamAppsSessionMetrics.java

```diff
@@ -0,0 +1,45 @@
+package org.teamapps.ux.session;
+
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.binder.MeterBinder;
+import org.teamapps.uisession.TeamAppsSessionManager;
+import org.teamapps.uisession.UiSessionState;
+
+public class TeamAppsSessionMetrics implements MeterBinder {
+
+    private final TeamAppsSessionManager sessionManager;
+
+    public TeamAppsSessionMetrics(TeamAppsSessionManager sessionManager) {
+        this.sessionManager = sessionManager;
+    }
+
+    @Override
+    public void bindTo(MeterRegistry registry) {
+
+        Gauge.builder("teamapps.session.ui.all", sessionManager, TeamAppsSessionManager::getNumberOfSessions)
+                .description("Total number of all registered UI sessions.")
+                .register(registry);
+
+        Gauge.builder("teamapps.session.ui.active", () -> sessionManager.getNumberOfSessionsByState(UiSessionState.ACTIVE))
+                .description("Total number of active UI sessions.")
+                .register(registry);
+
+        Gauge.builder("teamapps.session.ui.nearly_inactive", () -> sessionManager.getNumberOfSessionsByState(UiSessionState.NEARLY_INACTIVE))
+                .description("Total number of nearly inactive UI sessions.")
+                .register(registry);
+
+        Gauge.builder("teamapps.session.ui.inactive", () -> sessionManager.getNumberOfSessionsByState(UiSessionState.INACTIVE))
+                .description("Total number of inactive UI sessions.")
+                .register(registry);
```

Comment:
Use tags for subsets of sessions. sum of them should be all.

"_Total_" is normally used for counters. "_Current_ number of sessions" is a better description for gauges.

use `teamapps.uisessions` 

metric should look like: `teamapps_uisessions{dimension="inactive"}`

get sum of all sessions `sum(teamapps_uisessions{environment="PROD"})`